executor/aggfuncs/func_var_pop.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggfuncs
+
+import (
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pkg/errors"
+)
+
+// varianceKind stands for kind of population standard deviation/sample variance aggregate functions.
+type varianceKind int
+
+const (
+	kindNone varianceKind = iota
+	kindVariance
```

Comment:
considered there is only two variance kinds, maybe it's better to use the `isSample` predicate? For example, change the definition of `baseVarianceFloat64` to:
```go
type baseVarianceFloat64 struct {
	baseAggFunc
	isSample bool
}
```

Reply:
aggfuncs like STDDEV_POP/STDDEV_SAMP/VAR_POP/VAR_SAMP have a little diffentent in the final step. eg.
- VAR_POP -> variance/n
- VAR_SAMP -> variance/n-1
- STDDEV_POP -> sqrt(variance/n)
- STDDEV_SAMP -> sqrt(variance/n-1)

refer to Hive impl, I add varianceKind to distinguish different situations. Maybe I should add more kinds to avoid misunderstanding.

Reply:
OK.

Reply:
@lzmhhh123 the implementation of `std/stddev/stddev_pop/stddev_samp` in https://github.com/pingcap/tidb/pull/7766 should also be changed.